intake_stac/drivers.py

```diff
@@ -0,0 +1,217 @@
+        path_as_pattern=True,
+        storage_options=None,
+        **kwargs,
+    ):
+        self.path_as_pattern = path_as_pattern
+        self.urlpath = urlpath
+        self.chunks = chunks
+        self.dim = concat_dim
+        # self.storage_options = storage_options or {} #only relevant to fsspec?
+        self.override_coords = override_coords
+        self._kwargs = xarray_kwargs or {}
+        self._ds = None
+        # if isinstance(self.urlpath, list):
+        #    self._can_be_local = fsspec.utils.can_be_local(self.urlpath[0])
+        # else:
+        #    self._can_be_local = fsspec.utils.can_be_local(self.urlpath)
+
+        # Why is this necessary?
+        super(RioxarraySource, self).__init__(metadata=metadata)
+
+    def _open_items(self):
+        """
+        Use STAC metadata to intelligently concatenate multiple items
+        """
+        import xarray as xr
+
+        data_arrays = []
+        for item_id in self.metadata['items'].keys():
+            files = self.metadata['items'][item_id]['assets'].keys()
+            data_arrays.append(self._open_assets(files, item_id))
+
+        # by default concatenate items in time
+        ds = xr.concat(data_arrays, dim='item').swap_dims({'item': 'time'})
+        ds.name = None
+
+        return ds
+
+    def _open_assets(self, files, item_id=None):
+        """
+        use STAC metadata to intelligently concatenate multiple assets
+        """
+
+        import rioxarray
+        import xarray as xr
+
+        # Re-arranged metadata from intake-stac CombinedAssets
+        metadata = self.metadata['items'][item_id]
+
+        # not metadata-aware, so this assigns band=1 regardless of true band#
+        # Note: wrap with dask.delayed for parallel loading?
+        das = [rioxarray.open_rasterio(f, chunks=self.chunks, **self._kwargs) for f in files]
+        out = xr.concat(das, dim=self.dim)
+
+        # NOTE: no time zone conversion logic (assume UTC)
+        coords = {'item': item_id, 'time': metadata['STAC']['datetime'].replace(tzinfo=None)}
+
+        # by default assign asset keys as coordinate values
+        coords[self.dim] = [metadata['assets'][f]['key'] for f in files]
+
+        if self.pattern:
+            pattern_matches = reverse_formats(self.pattern, files)
+            coords[self.dim] = pattern_matches[self.dim]
+
+        if self.override_coords:
+            coords[self.dim] = self.override_coords
+
+        # copy item property metadata as attribute ?
+        # self._ds.attrs['STAC'] = self.metadata
+        out.name = item_id
+
+        return out.assign_coords(**coords)
+
+    def _open_dataset(self):
+        import rioxarray
+
+        # if self._can_be_local:
+        #    files = fsspec.open_local(self.urlpath, **self.storage_options)
+        # else:
+        # pass URLs to delegate remote opening to rasterio library
+        #    files = self.urlpath
+        # files = fsspec.open(self.urlpath, **self.storage_options).open()
```

Comment:
again i feel like none of this is needed unless the backend library knows what to do with fsspec things?:
```
with fsspec.open(url) as f:
     rioxarray.open_rasterio(f)
```

Reply:
Right, if we know for certain that we want original URLs passed through to the xarray backend, then fsspec has much less/nothing to do. It means there needs to be an alternate way to pass IO parameters (e.g., credentials) to that backend if it needs them.

Reply:
> It means there needs to be an alternate way to pass IO parameters (e.g., credentials) to that backend

I assume this means setting GDAL configs @scottyhq